libs/common/src/change-stream/delete-document-with-audit-logs.ts

```diff
@@ -0,0 +1,66 @@
+import { ObjectId, TransactionOptions } from 'mongodb';
+import { add } from 'date-fns';
+import { AuditDetails, MongoDbCollectionName } from '../types';
+import { MongoDbClient } from '../mongo-db-client';
+import { generateAuditDatabaseRecordFromAuditDetails } from './generate-audit-database-record';
+import { InvalidEnvironmentVariableError } from '../errors';
+
+type DeleteDocumentWithAuditLogsParams = {
+  documentId: ObjectId;
+  collectionName: MongoDbCollectionName;
+  db: MongoDbClient;
+  auditDetails: AuditDetails;
+};
+
+export const deleteDocumentWithAuditLogs = async ({
+  documentId,
+  collectionName,
+  db,
+  auditDetails,
+}: DeleteDocumentWithAuditLogsParams) => {
+  const { DELETION_AUDIT_LOGS_DELETE_AFTER_SECONDS } = process.env;
+
+  if (!DELETION_AUDIT_LOGS_DELETE_AFTER_SECONDS) {
+    throw new InvalidEnvironmentVariableError('DELETION_AUDIT_LOGS_DELETE_AFTER_SECONDS not set');
+  }
+
+  const client = await db.getClient();
+  const session = client.startSession();
+
+  try {
+    // TODO: verify that these are the options we want
```

Comment:
TODO

Reply:
https://www.mongodb.com/docs/manual/reference/read-concern/

Reply:
I think writeConcern `majority` is sensible

I think read concern shouldn't cause any issues in this transaction (since it only inserts and deletes) but it looks like the following options could make sense:
- `local` - only checks one instance, not guarantee that the data is consistent accross multiple replica set members
- `majority` - checks the data is consistent in the majority of members
- `snapshot` - like majority but only as it appeared from a specifiy point in time in the recent past (Note:  provides its guarantees only if the transaction commits with write concern "majority".)